test/test_dataloader.py

```diff
@@ -2262,10 +2276,31 @@ def test_shuffle(self):
         self.assertNotEqual(items, list(dl))
         self.assertEqual(items, sorted(list(dl)))
 
-        dl = DataLoader2(self.Sorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        dl = DataLoader2(self.IterSorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        self.assertEqual(list(dl), items)
+
+        dl = DataLoader2(self.IterSorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        self.assertEqual(list(dl), items)
+
+    def test_map_shuffle(self):
+        items = list(range(1000))
+        dp = SequenceWrapper(items).shuffle()
+
+        dl = DataLoader2(dp, batch_size=None, num_workers=2, shuffle=False)
+        self.assertEqual(items, list(dl))
+
+        dl = DataLoader2(dp, batch_size=None, num_workers=2, shuffle=True)
+        self.assertNotEqual(items, list(dl))
+        self.assertEqual(items, sorted(list(dl)))
+
+        dl = DataLoader2(dp, batch_size=None, num_workers=2, shuffle=True)
+        self.assertNotEqual(items, list(dl))
+        self.assertEqual(items, sorted(list(dl)))
+
+        dl = DataLoader2(self.MapSorter(dp), batch_size=None, num_workers=2, shuffle=True)
         self.assertEqual(list(dl), items)
 
-        dl = DataLoader2(self.Sorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        dl = DataLoader2(self.MapSorter(dp), batch_size=None, num_workers=2, shuffle=True)
```

Comment:
I can convert the `DataLoader2` to `DataLoader` to make it clear. This `DataLoader2` is actually used `DataLoader` internally: https://github.com/pytorch/pytorch/blob/9c5c045f63fc142a4d9e6e132912236f1dd2bc7d/torch/utils/data/dataloader_experimental.py#L99-L111

cc: @NivekT

Reply:
Let's do that since eventually they will be different (or we will remove the experimental from here).